### ISSUE TITLE
Should Octave.app be versioned like Octave-X.Y.Z.app?

### ISSUE THREAD

**Issue by apjanke** [octave-app/octave-app-bundler#4]
Should we version the installed Octave.app itself?

This would allow for side-by-side installations of multiple versions of Octave.app, at the cost of looking more "technical". I don't know how much use case there is for this.

For users, if they are conservative and want to test their Octave code against multiple versions of Octave (e.g. before adopting a new update for production use), this would let them install the old version and the tentative new version side-by-side, and run both on their computer.

It would also be useful for Octave.app maintainers, letting them install prerelease versions side by side with release versions. (If we're lucky, maybe we could get some users to test prereleases, too.)

Matlab does its releases like this: the apps are named `MATLAB_R2017b`, `MATLAB_R2018a`, and so on. I've found that useful in my own work: I work on different projects with different compatibility levels, and also write some code that needs to have compatibility spanning multiple versions of Matlab, so it's essential to have them installed side-by-side on my dev machines. The same might be true for Octave users.

**Comment by apjanke** [comment]
I'm leaning towards Yes on this, if only because that will allow devs and users to easily install and test stable, development, and prerelease versions of Octave side-by-side. Without this, they either need to do a clumsy manual rename-based management of versions, or we need to provide a version-switcher mechanism (ugh).

One other benefit that is available _only_ with versioned apps: testers could run a stable and prerelease version of Octave.app side-by-side at the same time, comparing their behavior. Otherwise, they'd need to run one, record the results, switch out to the other version, run their code, and compare it to the recorded results. Given the complex functionality of Octave, I think that might be important.

**Comment by apjanke** [comment]
Actually, I'm leaning really strongly towards Yes on this, because the current Octave.app binaries are non-relocatable, so there's no way for users to manage side-by-side installations themselves, aside from swapping files in and out, which is icky.

**Comment by apjanke** [comment]
Do you think there's a use case for an OpenBLAS or similar variant build of Octave.app, too? If so, we should also include variant builds somewhere in the naming scheme. (Regular build gets no qualifier; variant builds have something like "-openblas" stuck in the name, probably before the version.)

**Comment by schoeps** [comment]
Yes. I had the same thought. If we manage to get the relative pathes in the libs done, then people can rename the bundle for beautification of their App-folder :-)
